File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/teleop/MotorTester.java
package org.firstinspires.ftc.teamcode.teleop;

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.Gamepad;

@TeleOp(name = "Motor Tester")
public class MotorTester extends LinearOpMode {
    @Override
    public void runOpMode() {
        DcMotor horSlide = hardwareMap.get(DcMotor.class, "verticalSlide");
        horSlide.setTargetPosition(0);
        horSlide.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
        horSlide.setMode(DcMotor.RunMode.RUN_TO_POSITION);
        waitForStart();


        while (opModeIsActive()) {
            if (gamepad1.aWasPressed()) {
                horSlide.setTargetPosition(horSlide.getTargetPosition()-25);
            }

            if (gamepad1.yWasPressed()) {
                horSlide.setTargetPosition(horSlide.getTargetPosition()+25);
            }

            horSlide.setPower(1.0);

            telemetry.addLine("Running");
            telemetry.addLine("Encoder Position: " + horSlide.getCurrentPosition());
            telemetry.addData("Target Position:", horSlide.getTargetPosition());
            telemetry.addData("Motor speed:", horSlide.getPower());
            telemetry.update();

            sleep(50);
        }
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/teleop/MotorTester.kt
package org.firstinspires.ftc.teamcode.teleop

import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.Methods

@TeleOp(name = "Motor Tester")
class MotorTester : Methods() {
    override fun runOpMode() {

        initMotors()
        waitForStart()

        while (opModeIsActive()) {
            val driverCurrent = gamepad1
            val driverPrevious = gamepad1
            val gunnerCurrent = gamepad2
            val gunnerPrevious = gamepad2

            driverCurrent.copy(driverPrevious)
            gamepad1.copy(driverCurrent)

            gunnerCurrent.copy(gunnerPrevious)
            gamepad2.copy(gunnerCurrent)

            val up = gamepad1.y
            val down = gamepad1.a

            vertSlide!!.power = if (up) 0.1 else if (down) -0.1 else 0.0

            telemetry.addLine("Running")
            telemetry.addLine("Encoder Position: ${vertSlide!!.currentPosition}")
            telemetry.update()
        }
    }
}